fabric-events-lifecycle-v0/src/main/java/net/fabricmc/fabric/api/event/server/ServerReloadCallback.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.server;
+
+import net.minecraft.server.MinecraftServer;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * A callback for the server's reloading, as of {@code /reload} command is typed.
```

Comment:
I assume mods could trigger a reload as well.
```suggestion
 * A callback for the when the server reloads, such as when the {@code /reload} command is typed.
```

Reply:
Note that `/reload` isn't a server reload, it's a server resource reload.

Reply:
@Earthcomputer certain elements beyond server resources are reloaded, e.g. player data is saved and loaded again.

> what's an example of how a mod would use it?

A mod would temporarily transfer its data like ids to some error-acceptable format (identifiers instead of object references to prevent resource leaks etc.) and then convert data back to in-memory format after the reload.